litgpt/eval/evaluate.py

```diff
@@ -0,0 +1,114 @@
+# Copyright Lightning AI. Licensed under the Apache License 2.0, see LICENSE file.
+
+import json
+import os
+from pathlib import Path
+from typing import Optional
+import yaml
+import torch
+
+from litgpt.scripts.convert_lit_checkpoint import convert_lit_checkpoint
+from litgpt.utils import CLI, copy_config_files
+
+
+def safe_safetensors(out_dir, repo_id):
+    from transformers import AutoModel
+
+    state_dict = torch.load(out_dir/"model.pth")
+    model = AutoModel.from_pretrained(
+        repo_id, state_dict=state_dict
+    )
+    model.save_pretrained(out_dir)
```

Comment:
Do we really need to save it here? Does the lm eval harness not support taking a HF model instance instead?

Reply:
You mean a model instance generated via `[convert_lit_checkpoint.py](https://github.com/Lightning-AI/litgpt/blob/main/litgpt/scripts/convert_lit_checkpoint.py)`? This usually never worked with the harness. I think @awaelchli also tried that before without success.

Reply:
I mean using the `model` instance you have in L18 which according to their docs can be passed in https://github.com/EleutherAI/lm-evaluation-harness/blob/main/lm_eval/evaluator.py#L61-L62

The advantage being that you skip the step of saving another copy of the weights

Reply:
I see. I tried that (see current PR) but having issues with this. Now resulting in a 

```
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 235/235 [00:00<00:00, 816.08it/s]
2024-03-27:17:55:45,864 INFO     [task.py:395] Building contexts for truthfulqa_mc2 on rank 0...
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 817/817 [00:00<00:00, 981.97it/s]
2024-03-27:17:55:46,750 INFO     [task.py:395] Building contexts for gsm8k on rank 0...
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1319/1319 [00:05<00:00, 229.79it/s]
2024-03-27:17:55:52,516 INFO     [task.py:395] Building contexts for hellaswag on rank 0...
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 10042/10042 [00:03<00:00, 3273.56it/s]
2024-03-27:17:55:56,577 INFO     [evaluator.py:362] Running loglikelihood requests
Running loglikelihood requests:   0%|                                                                                                                                                  | 0/109439 [00:00<?, ?it/s]Traceback (most recent call last):
  File "/home/zeus/miniconda3/envs/cloudspace/bin/litgpt", line 8, in <module>
    sys.exit(main())
  File "/teamspace/studios/this_studio/litgpt/litgpt/__main__.py", line 131, in main
    fn(**kwargs)
  File "/teamspace/studios/this_studio/litgpt/litgpt/eval/evaluate.py", line 100, in convert_and_evaluate
    results = evaluator.simple_evaluate(
  File "/home/zeus/miniconda3/envs/cloudspace/lib/python3.9/site-packages/lm_eval/utils.py", line 288, in _wrapper
    return fn(*args, **kwargs)
  File "/home/zeus/miniconda3/envs/cloudspace/lib/python3.9/site-packages/lm_eval/evaluator.py", line 234, in simple_evaluate
    results = evaluate(
  File "/home/zeus/miniconda3/envs/cloudspace/lib/python3.9/site-packages/lm_eval/utils.py", line 288, in _wrapper
    return fn(*args, **kwargs)
  File "/home/zeus/miniconda3/envs/cloudspace/lib/python3.9/site-packages/lm_eval/evaluator.py", line 373, in evaluate
    resps = getattr(lm, reqtype)(cloned_reqs)
  File "/home/zeus/miniconda3/envs/cloudspace/lib/python3.9/site-packages/lm_eval/api/model.py", line 329, in loglikelihood
    return self._loglikelihood_tokens(new_reqs, disable_tqdm=disable_tqdm)
  File "/home/zeus/miniconda3/envs/cloudspace/lib/python3.9/site-packages/lm_eval/models/huggingface.py", line 1033, in _loglikelihood_tokens
    self._model_call(batched_inps, **call_kwargs), dim=-1
  File "/home/zeus/miniconda3/envs/cloudspace/lib/python3.9/site-packages/lm_eval/models/huggingface.py", line 748, in _model_call
    return self.model(inps).logits
AttributeError: 'BaseModelOutputWithPast' object has no attribute 'logits'
```

Not sure how to fix that, I find it hard to figure out what I am missing. Unless it's something obvious you maybe know how to fix, I suggest reverting back to the state before and save the converted model even though it's wasteful. We can then open an issue and reinvestigate that later and see if we find a way to evaluate for the HF object directly. This would not change the API or anything, so I would say let's leave that as an optional thing?